CMakeLists.txt

```diff
@@ -1,125 +1,72 @@
 cmake_minimum_required(VERSION 2.8)
+
 project(fccedm CXX)
+set(fccedm_MAJOR_VERSION 1)
+set(fccedm_MINOR_VERSION 0)
+set(fccedm_PATCH_VERSION 0)
+set(fccedm_VERSION
+  ${fccedm_MAJOR_VERSION}.${fccedm_MINOR_VERSION}.${fccedm_PATCH_VERSION})
 
-set(FCCEDM_MAJOR_VERSION 1)
-set(FCCEDM_MINOR_VERSION 0)
-set(FCCEDM_PATCH_VERSION 0)
-set(FCCEDM_VERSION
-  ${FCCEDM_MAJOR_VERSION}.${FCCEDM_MINOR_VERSION}.${FCCEDM_PATCH_VERSION})
+#--- Test if this is actually needed
+if(APPLE)
+  set(CMAKE_SHARED_LIBRARY_SUFFIX ".so")
+endif(APPLE)
 
 #--- Declare options -----------------------------------------------------------
 option(fccedm_documentation "Whether or not to create doxygen doc target.")
 
-# Offer the user the choice of overriding the installation directories
+#--- set up default install dirs -----------------------------------------------
 set(INSTALL_LIB_DIR lib CACHE PATH "Installation directory for libraries")
 set(INSTALL_BIN_DIR bin CACHE PATH "Installation directory for executables")
 set(INSTALL_INCLUDE_DIR include CACHE PATH
   "Installation directory for header files")
 
+#-- set compiler flags ---------------------------------------------------------
+add_definitions(-Wno-unused-variable -Wno-unused-parameter -pthread)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -DDROP_CGAL -std=c++11 -Wall -Wextra -Wpedantic")
```

Comment:
Which do we want to set? They were set several times to different values before...